addons/stock/models/product.py

```diff
@@ -270,7 +271,8 @@ def _search_ids(model, values, force_company_id):
         warehouse = self.env.context.get('warehouse')
         if warehouse and not isinstance(warehouse, list):
             warehouse = [warehouse]
-        force_company = self.env.context.get('force_company', False)
+        force_company = self.env.cid or False
```

Comment:
Totally remove force_company logic.
Are those product quantities accessed in sudo anyway ?

Reply:
Kept at sle demand, could be useful.